File: frontend/src/types/session.ts
import type { TableData } from './chart'

export type Role = 'user' | 'assistant'

/** 与后端 `GET /api/sessions` 单条记录（SessionOut）字段一致（snake_case） */
export interface SessionDTO {
  id: string
  title: string
  created_at: number
  updated_at: number
}

/**
 * 与后端 `GET /api/sessions/{id}/messages`（MessageOut）一致；
 * `GET .../messages` 无顶层 `table` 字段：若助手落库时在 `assistant_meta.table` 中保存预览数据，前端会填入 `ChatThreadMessage.table`。
 */
export interface ChatThreadMessage {
  id: string
  session_id?: string
  role: Role
  content: string
  sql_text?: string | null
  assistant_meta?: Record<string, unknown> | null
  created_at?: number
  table?: TableData | null
  /** 与 `assistant_meta.viz_insight` 同步；落库后刷新会话仍可从 meta 恢复 */
  viz_insight?: string | null
}

File: backend/app/schemas/session.py
"""会话与历史消息契约。"""

from __future__ import annotations

from typing import Optional

from pydantic import BaseModel, Field


class SessionCreate(BaseModel):
    title: str = Field(default="新会话", min_length=1, max_length=200)


class SessionPatch(BaseModel):
    title: str = Field(min_length=1, max_length=200)


class SessionOut(BaseModel):
    model_config = {"from_attributes": True}

    id: str
    title: str
    created_at: int
    updated_at: int


class MessageOut(BaseModel):
    model_config = {"from_attributes": True}

    id: str
    session_id: str
    role: str
    content: str
    sql_text: Optional[str] = None
    assistant_meta: Optional[dict] = None
    created_at: int


class VizInsightOut(BaseModel):
    viz_insight: str
